virtual_ecosystem/core/registry.py

```diff
@@ -184,6 +192,46 @@ def register_module(module_name: str) -> None:
                 class_name,
             )
 
+    # Find and register the model configuration
+    try:
+        config_submodule = import_module(f"{module_name}.model_config")
+    except ModuleNotFoundError:
+        raise RuntimeError("Model does not provide a model_config submodule.")
+
+    # Get all subclasses inheriting from ConfigRoot from the model_config submodule
+    # This includes the imported base classes themselves, and also nested
+    # ModelConfigSection subclasses, which need to be filtered out.
+    model_config_subclasses = {
+        class_name: class_obj
+        for class_name, class_obj in getmembers(config_submodule)
+        if isclass(class_obj)
+        and issubclass(class_obj, ConfigRoot)
+        and not issubclass(class_obj, ModelConfigSection)
+        and class_obj is not ConfigRoot
+        and class_obj is not ModelConfigRoot
+    }
```

Comment:
Not sure if it has any advantage, but you can also do:

```suggestion
    model_config_subclasses = dict(
        getmembers(
            config_submodule, 
            lambda obj: (
                if isclass(obj)
                and issubclass(obj, ConfigRoot)
                and not issubclass(obj, ModelConfigSection)
                and obj is not ConfigRoot
                and obj is not ModelConfigRoot
            )
        )
    )
```

Reply:
Thinking about it, I wonder if it won't be better to just enforce a naming convention related to the model name instead of "searching" for a single config object. So, something like the following (in a separate function, otherwise this becomes enormous):

```python
def get_config_class(module_name):
    name = ''.join(word.capitalize() for word in module_name.split("_")) + "Config"

    try:
        config_submodule = import_module(f"{module_name}.model_config")
        model_config_class = getattr(config_submodule, name)
        assert issubclass(model_config_class, ModelConfigRoot)
    except ModuleNotFoundError:
        raise RuntimeError("Model does not provide a model_config submodule.")
    except AttributeError:
        raise RuntimeError(
            f"Root config class {name} not found for model {module_name} in its `model_config` module."
        )
    except AssertionError:
        raise RuntimeError(
            f"Model {module_name} config class does does inherit from `ConfigRoot`."
        )

    return model_config_class
```

Or something along these lines. I feel this is cleaner.

Reply:
I think that's right. 

I'd probably go further and not even have a `module_name` component. Model developers have to provide `xyz_model.model_config.ModelConfiguration` as the root configuration class. That's what I've implemented in the edits.